### PR TITLE
add missing killSwitch for parallel outbound lanes, #21381

### PR DESCRIPTION
**Pull request by patriknw**
- it caused the shutdown to stall, since the part after MergeHub
  was never stopped

Refs  #21381

### CODE REVIEW

File: akka-remote/src/main/scala/akka/remote/artery/Association.scala [hunk 1]
```diff
@@ -473,7 +473,9 @@ private[remote] class Association(
           case (((q, m), c), w) ⇒ ((q, m), (c, w))
         }
 
-      val (mergeHub, aeronSinkCompleted) = MergeHub.source[EnvelopeBuffer].toMat(transport.aeronSink(this))(Keep.both).run()(materializer)
+      val (mergeHub, aeronSinkCompleted) = MergeHub.source[EnvelopeBuffer]
+        .via(transport.killSwitch.flow[EnvelopeBuffer])
```
Comment (patriknw) [inline akka-remote/src/main/scala/akka/remote/artery/Association.scala]: this is not enough, I also need to make sure that all parts are teared down if one of them fails

**Comment by patriknw** [comment]
updated this to handle tear down of parts downstream and upstream of the hub together

**Comment by johanandren** [comment]
LGTM

**Comment by drewhk** [comment]
LGTM.

**Pull request merged**